cpp/include/cudf/join.hpp

```diff
@@ -785,7 +781,8 @@ std::pair<std::unique_ptr<rmm::device_uvector<size_type>>,
 conditional_full_join(table_view left,
                       table_view right,
                       ast::expression binary_predicate,
-                      null_equality compare_nulls         = null_equality::EQUAL,
+                      null_equality compare_nulls            = null_equality::EQUAL,
+                      std::optional<std::size_t> output_size = {},
```

Comment:
Does it make sense to preserve this parameter if there's no API to compute it?

Reply:
Good point, we can remove it for now. Probably better than leaving it purely for symmetry with the other joins.